generate/aff.mjs

```diff
@@ -31,38 +31,70 @@ async function run() {
 
 
   const types = [];
-  for (const f of fs.readdirSync("../file-formats/")) {
+  const directory = '../file-formats'; // Replace with your directory
+  for (const f of fs.readdirSync(directory)) {
     if (f.length === 4) {
-      types.push(f.toUpperCase());
+      types.push(f.toLowerCase());
     }
   }
   types.sort();
 
 
   let error = false;
   for (const type of types) {
-    if (type === "ENHO") {
-      core.notice(type + " skipped, https://github.com/SAP/abap-file-formats/issues/409");
+    if (type.toUpperCase() === "ENHO") {
+      core.notice(type.toUpperCase() + " skipped, https://github.com/SAP/abap-file-formats/issues/409");
       continue;
     }
 
-    const result = await abap.Classes["CL_RUN"].run({ object_type: new abap.types.String().set(type) });
-    const filename = "generated" + path.sep + type.toLowerCase() + "-v1.json";
-    const filename_aff = `../file-formats/${type.toLowerCase()}/${type.toLowerCase()}-v1.json`;
-    fs.writeFileSync(filename, result.get());
-
-
-    const command = `diff ${filename_aff} ${filename}`;
-    const output = child_process.execSync(`${command} || true`);
-    if (output.toString().length > 0) {
-      core.setFailed(type + ": Provided and generated JSON Schema differ")
-      createAnnotations(output.toString(), path.resolve(filename_aff));
-      //core.info(command);
-      //core.info(output.toString());
-    } else {
-      core.notice(type + " success");
+    const files = fs.readdirSync(directory + path.sep + type);
+    const jsonFiles = files.filter(file => path.extname(file) === '.json');
+    let objTypeVersNumb = jsonFiles.map((file) => {
+      // Extract the characters before '-v' as namePart and the number after '-v' as versionNumber using regex
+      let match = file.match(/(.*)(?:-v)(\d+)(?=\.json)/);
```

Comment:
Later, when the functionality might generate the schema, I guess we cannot use this approach to identify the version number (because the file might not exist, yet). However, for now it seems to be fine to me.